connectors/redis/redis_test.go

```diff
@@ -0,0 +1,215 @@
+package redis_test
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber-go/dosa"
+	"github.com/uber-go/dosa/connectors/redis"
+	"github.com/uber-go/dosa/testentity"
+	"golang.org/x/net/context"
+)
+
+var testRedisConfig = redis.Config{
+	ServerSettings: redis.ServerConfig{
+		Host: "localhost",
+		Port: redis.RedisPort,
+	},
+	TTL: 5 * time.Second,
+}
+
+var rc = redis.NewConnector(testRedisConfig)
+
+func TestUnimplementedFunctions(t *testing.T) {
+	err := rc.CreateIfNotExists(context.TODO(), &dosa.EntityInfo{}, nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.MultiRead(context.TODO(), &dosa.EntityInfo{}, nil, dosa.All())
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.MultiUpsert(context.TODO(), &dosa.EntityInfo{}, nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	err = rc.RemoveRange(context.TODO(), &dosa.EntityInfo{}, nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.MultiRemove(context.TODO(), &dosa.EntityInfo{}, nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, _, err = rc.Range(context.TODO(), &dosa.EntityInfo{}, nil, dosa.All(), "", 1)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, _, err = rc.Search(context.TODO(), &dosa.EntityInfo{}, dosa.FieldNameValuePair{}, dosa.All(), "", 1)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, _, err = rc.Scan(context.TODO(), &dosa.EntityInfo{}, dosa.All(), "", 1)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.CheckSchema(context.TODO(), "", "", nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.UpsertSchema(context.TODO(), "", "", nil)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.CheckSchemaStatus(context.TODO(), "", "", 0)
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	err = rc.CreateScope(context.TODO(), "")
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	err = rc.TruncateScope(context.TODO(), "")
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	err = rc.DropScope(context.TODO(), "")
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	_, err = rc.ScopeExists(context.TODO(), "")
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+
+	err = rc.Shutdown()
+	assert.EqualError(t, err, new(redis.ErrNotImplemented).Error())
+}
+
+func TestWriteReadKeyValue(t *testing.T) {
+	if !redis.IsRunning() {
+		t.Skip("Redis is not running")
+	}
+
+	table, _ := dosa.TableFromInstance(&testentity.KeyValue{})
+	sr := dosa.SchemaRef{Scope: "example", NamePrefix: "example"}
+	testEi := &dosa.EntityInfo{Ref: &sr, Def: &table.EntityDefinition}
+	values := map[string]dosa.FieldValue{
```

Comment:
nit: these keep repeating so perhaps they should be declared outside of all the tests